File: src/main/dotNet/GlobalCode.Surfing/GlobalCode.Surfing/Device/IoTSurfboard.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace GlobalCode.Surfing.Device
{

    /// <summary>
    /// API de comunicação com a IoTSurfboard
    /// </summary>
    public class IoTSurfboard
    {
        private readonly IDevice _board;

        /// <summary>
        /// API de comunicação com a IoTSurfboard
        /// </summary>
        /// <param name="port">Porta de comunicação Ex: COM5</param>
        /// <param name="baudRate">Velocidade de comunicação Ex: 9600</param>
        public IoTSurfboard(string port, int baudRate)
        {
            _board = new SerialDeviceRxTx(port, baudRate);
            _board.Open();
            Thread.Sleep(2500);
        }

        /// <summary>
        /// Finaliza a comunicação com a IoTSurfboard
        /// </summary>
        public void Close()
        {
            try
            {
                _board.Close();
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Retorna o valor do sensor de Alcool
        /// </summary>
        /// <returns>Valor 0-1023 do sensor</returns>
        public int Alcohol()
        {
            try
            {
                _board.Send("alcohol");
                Thread.Sleep(100);
                return int.Parse(_board.Receive());
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Retorna valor do sensor de luz
        /// </summary>
        /// <returns>Valor 0-1023 do sensor</returns>
        public int Light()
        {
            try
            {
                _board.Send("light");
                Thread.Sleep(100);
                return int.Parse(_board.Receive());
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Retorna valor do indicador do potenciômetro
        /// </summary>
        /// <returns>Valor 0-1023 do sensor</returns>
        public int Potentiometer()
        {
            try
            {
                _board.Send("pot");
                Thread.Sleep(100);
                return int.Parse(_board.Receive());
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Retorna a temperatura
        /// </summary>
        /// <returns>Temperatura Celsius</returns>
        public float Temperature()
        {
            try
            {
                _board.Send("temp");
                Thread.Sleep(350);
                return float.Parse(_board.Receive());
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Retorna o valor do sensor de umidade
        /// </summary>
        /// <returns>Porcentagem da umidade</returns>
        public float Humidity()
        {
            try
            {
                _board.Send("humidity");
                Thread.Sleep(350);
                return float.Parse(_board.Receive());
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Controla o Transistor
        /// </summary>
        /// <param name="v">true = liga / false = desliga</param>
        public void Transistor(bool v)
        {
            try
            {
                _board.Send("transistor?" + (v ? "1" : "0"));
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Retorna a Data/Hora do Device
        /// </summary>
        /// <returns>Data / Hora</returns>
        public string Clock()
        {
            try
            {
                _board.Send("clock");
                Thread.Sleep(100);
                return _board.Receive();
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Controle o relê do device
        /// </summary>
        /// <param name="v">true = liga / false = desliga</param>
        public void Relay(bool v)
        {
            try
            {
                _board.Send("relay?" + (v ? "1" : "0"));
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Controle o Speaker
        /// </summary>
        /// <param name="v">true = liga / false = desliga</param>
        public void Speaker(bool v)
        {
            try
            {
                _board.Send("speaker?" + (v ? "1" : "0"));
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }

        }

        /// <summary>
        /// Aciona o Led Vermelho
        /// </summary>
        /// <param name="p">Itensidade (0=desliga)</param>
        public void Red(int p)
        {
            try
            {
                _board.Send("red?" + p);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Aciona o Led Verde
        /// </summary>
        /// <param name="p">Itensidade (0=desliga)</param>
        public void Green(int p)
        {
            try
            {
                _board.Send("green?" + p);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Aciona o Led Azul
        /// </summary>
        /// <param name="p">Itensidade (0=desliga)</param>
        public void Blue(int p)
        {
            try
            {
                _board.Send("blue?" + p);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

        /// <summary>
        /// Controla o Led RGB
        /// </summary>
        /// <param name="r">Itensidade Vermelho (0=desliga)</param>
        /// <param name="g">Itensidade Verde (0=desliga)</param>
        /// <param name="b">Itensidade Azul (0=desliga)</param>
        public void Rgb(int r, int g, int b)
        {
            try
            {
                this.Red(r);
                Thread.Sleep(50);

                this.Green(g);
                Thread.Sleep(50);

                this.Blue(b);
                Thread.Sleep(50);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }

        }


        /// <summary>
        /// Controla do Servo (Opcional)
        /// </summary>
        /// <param name="p">Posição</param>
        public void Servo(int p)
        {
            try
            {
                _board.Send("servo?" + p);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }

        }

        /// <summary>
        /// Retorna o valor do sensor de distância
        /// </summary>
        /// <returns>valor sensor de distância em CM</returns>
        public int Distance()
        {
            try
            {
                _board.Send("distance");
                Thread.Sleep(150);
                return int.Parse(_board.Receive());
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex);
            }
        }

    }
}

File: src/main/java/org/surfing/device/IoTSurfboard.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package org.surfing.device;

import java.io.IOException;
import org.surfing.Device;
import org.surfing.kernel.Kernel;

/**
 *
 * @author vinic_000
 */
public class IoTSurfboard {

    Device board;

    public IoTSurfboard(String port, int baudRate) throws IOException {
        board = new SerialDeviceJSSC(port, baudRate);
        board.open();
        Kernel.delay(2500);
    }

    public IoTSurfboard(String port, int baudRate, String API) throws IOException {
        if (API.equals("RXTX")) {
            board = new SerialDeviceRXTX(port, baudRate);
        } else if (API.equals("JSSC")) {
            board = new SerialDeviceJSSC(port, baudRate);
        } else {
            System.out.println("API Name not recognized!");
        }
        board.open();
        Kernel.delay(2500);
    }

    public void close() throws IOException {
        board.close();
    }

    public int alcohol() throws IOException {
        board.send("alcohol");
        Kernel.delay(100);
        return Integer.parseInt(board.receive());
    }

    public int light() throws IOException {
        board.send("light");
        Kernel.delay(100);
        return Integer.parseInt(board.receive());
    }

    public int potentiometer() throws IOException {
        board.send("pot");
        Kernel.delay(100);
        return Integer.parseInt(board.receive());

    }

    public float temperature() throws IOException {
        board.send("temp");
        Kernel.delay(350);
        return Float.parseFloat(board.receive());

    }

    public float humidity() throws IOException {
        board.send("humidity");
        Kernel.delay(350);
        return Float.parseFloat(board.receive());
    }

    public void transistor(boolean v) throws IOException {
        board.send("transistor?" + (v ? "1" : "0"));
    }

    public String clock() throws IOException {
        board.send("clock");
        Kernel.delay(100);
        return board.receive();
    }

    public void relay(boolean v) throws IOException {
        board.send("relay?" + (v ? "1" : "0"));
    }

    public void speaker(boolean v) throws IOException {
        board.send("speaker?" + (v ? "1" : "0"));

    }

    public void red(int p) throws IOException {
        board.send("red?" + p);

    }

    public void green(int p) throws IOException {
        board.send("green?" + p);

    }

    public void blue(int p) throws IOException {
        board.send("blue?" + p);

    }

    public void rgb(int r, int g, int b) throws IOException {
        this.red(r);
        Kernel.delay(50);

        this.green(g);
        Kernel.delay(50);

        this.blue(b);
        Kernel.delay(50);

    }

    public void servo(int s) throws IOException {
        board.send("servo?" + s);

    }

    public int distance() throws IOException {
        board.send("distance");
        Kernel.delay(150);
        return Integer.parseInt(board.receive());

    }
}
